FindMe/Parse Manager/DatabaseManager.m

```diff
@@ -0,0 +1,86 @@
+//
+//  DatabaseManager.m
+//  FindMe
+//
+//  Created by Nicholas Wayoe on 7/21/20.
+//  Copyright © 2020 Nicholas Wayoe. All rights reserved.
+//
+
+#import "DatabaseManager.h"
+#import "Parse/Parse.h"
+
+@implementation DatabaseManager
+
++ (void)createUser:(NSDictionary *)userDetails withCompletion:(void(^)(NSError *error))completion
+{
+    PFUser *user = [PFUser new];
+    user.username = userDetails[@"username"];
+    user.password = userDetails[@"password"];
+    user[@"firstName"] = userDetails[@"firstName"];
+    user[@"lastName"] = userDetails[@"lastName"];
+    if (userDetails[@"profilePhotoData"]) {
+        user[@"profilePhoto"] = [PFFileObject fileObjectWithData:userDetails[@"profilePhotoData"]];
+    }
+    else {
+        
+    }
+    [user signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
+        if (error) {
+            completion(error);
+        }
+        else
+            completion(nil);
+    }];
+}
+
++ (void)getUser:(void(^)(PFUser * user))completion
+{
+    PFUser *user = [PFUser currentUser];
+    if(user) {
+        completion(user);
+    }
+    else {
+        completion(nil);
+    }
+}
+
++ (void)fetchContacts:(void(^)(NSArray *contacts, bool gotContacts))completion
```

Comment:
I always thought this objc pattern was sort of weird. If contacts is nil then `gotContacts` is false. This duplicates the number of cases a caller needs to handle without providing new information.

Reply:
Yeah I tried using just one case - contacts being nil or not but the compiler kept running the unexpected resuts